### DISCUSSION TITLE
Error loading tags: java.lang.StackOverflowError java.lang.StackOverflowError at java.desktop/java.awt.AWTEventMulticaster.remove(Unknown Source)

### DISCUSSION THREAD

**Discussion by tstac0427** [nextgenhealthcare/connect#5926] [Q&A]
I am seeing this in Mirth version 4.1.1, 4.3 and the latest version 4.4
Please let me know how to resolve this issue.
Has anyone else experienced similar error

**Comment by pacmano1** [comment]
You should provide more context.  When/where does this happen?  Screenshots? 

What java version on your local workstation andt the server?   

Please don't post error messages alone, provide details.

**Reply by JoeFox82** [reply]
Hi pacmano1,

it looks like this:
![mirth_client](https://github.com/nextgenhealthcare/connect/assets/96169329/9766a1ca-3c38-4327-9d93-75676cc69d45)

I have the same issue with version 4.4.x when the client is running longer time.

**Comment by tstac0427** [comment]
When this happens - I do not have exact timings but we have seen this 3 times in a week
Where it happens - Do you mean Test environment or Production ? - It happens in PROD, have shared below the prod server configurations
Screenshots - I do not  have one at this point but will capture it when it shows up again, just noted down the error 

Server 1
Java version - 8
Mirth version - 4.3.0

Server 2
Java version - 8
Mirth version - 4.1.1

Server 3
Java version - 15
Mirth version - 4.4.0

**Reply by pacmano1** [reply]
Where it happens = presumably on the mirth admin tool console log?  The java console?   A popup?

**Reply by kirbykn2** [reply]
Could this be a client side issue?  What JRE are you using?  What are your settings for the MCAL?

**Comment by tstac0427** [comment]
the popup is coming in the mirth dashboard.
We are using open source JDK as I mentioned the java version 8, jre version 8 and where Java version is 15 we use JRE 15.
What is MCAL ? Could you please elaborate

Does this answer your question on JRE?

**Reply by pacmano1** [reply]
MCAL = Mirth Connect Administrator Launcher.

**Comment by tstac0427** [comment]
this is the screenshot for one of the server MCAL
![MCAL](https://github.com/nextgenhealthcare/connect/assets/132012211/b8180cc5-84c1-40f6-ae0e-fe1e64865dc0)
On this server mirth connect uses Java version 8 (open source JDK) 
All 3 servers are Windows based.

**Reply by kirbykn2** [reply]
For funzies, assign more memory to the heap space, and try it again.  Do you have customer libraries loaded in Mirth connect?

**Reply by tstac0427** [reply]
On Server 1 we do not have any library
Server 2 - has a custom library
Server 3 - does not have any library

**Comment by ShraddhaGS** [comment]
Hi, What is the solution to the above problem as even I am facing the same issue. The error gets resolved only when Mirth is restarted.
Is there any solution for the issue? What is causing the issue in the first place?
I am also following https://forums.mirthproject.io/forum/mirth-connect/support/183947-stackoverflow-error

**Reply by ArdnasM** [reply]
Same thing here... 
Emptying the Java cache also appears to work, but only temporarily.

**Reply by JoeFox82** [reply]
After updating from Java 8 to Java 21 (used by the server not the MCAL) on our test server I didn't get this error anymore till now.

**Comment by puresimmer** [comment]
Did anyone ever find a definite answer to this?

Thanks!

**Reply by JoeFox82** [reply]
After updating our servers to Java 21 (used by the server not the MCAL) I didn't get this error anymore. Tested with the Mirth versions 4.4.0, 4.4.1, 4.4.2 and 4.5.0

**Comment by prc9584** [comment]
I started seeing this when updating devices to 4.5.0, and have seen with both java 8 and java 17.

**Reply by JoeFox82** [reply]
After updating our servers to Java 21 (used by the server not the MCAL) I didn't get this error anymore. Tested with the Mirth versions 4.4.0, 4.4.1, 4.4.2 and 4.5.0

**Reply by prc9584** [reply]
Going to try version 22 and see if it resolves.

**Reply by pacmano1** [reply]
@JoeFox82 so you are using MCAL on some other machine (not running mirth server) with either java 8 or 17 that ships with MCAL?  And updating the java version on the mirth server itself to 21 fixed he issue on the Arm macs?

**Reply by prc9584** [reply]
Note, previously this sometime took having MCA open for 24 hours or longer to initiate the error, it also sometimes happened almost immediately. I have updated the server MCA and MCAL are installed on to java v22, and kept MCAL set to launch OpenJDK 17, so far in about 2 hours no new error pop-up but will do a 24 hour report as well to see if either of the sample servers are effected again.

**Reply by JoeFox82** [reply]
I'm using MCAL with the JDK17 bundle within the installer (no Java 8 bundle). MC server is running with Java 21.

**Reply by prc9584** [reply]
Java 22 24 hours of monitoring and still no new StackOverflowError errors occurred. Will write a note if anything changes, but so far looks like a good correlation of at minimum java 21 to be installed to not get the errors indicated at the begining of this thread.

**Reply by prc9584** [reply]
Error came back over the weekend. No changes done on server.

**Reply by JoeFox82** [reply]
Today, one of our customers said, that he has the same error with Java 21.0.1 and Mirth 4.4.2. We've updated to Mirth 4.5.0 and now we are waiting if the error happens again. If it will happen again I'll try to update to Java 21.0.3 because Java 21.0.2 has two AWT fixes: https://www.oracle.com/java/technologies/javase/21-0-2-relnotes.html

**Reply by prc9584** [reply]
@JoeFox82 did you have any luck? Still seeing the issue on 4.5.0.

**Reply by JoeFox82** [reply]
After we updated the customer's Mirth to 4.5.0, the error no longer occurred. Java is still installed in version 21.0.1.

Edit 1: It happend again this morning.
Edit 2: It also happens with Mirth 4.4.2 and Java 21.0.3

**Reply by pcskatja** [reply]
Does anyone has a solution for this problem? We have now also a customer with this problem (Mirth 4.4.2)
@JoeFox82: is it solved with mirth 4.5.0?

**Reply by JoeFox82** [reply]
It also happens with Mirth 4.5.0 and Java 21.0.3. Could it be a windows session problem?

**Reply by prc9584** [reply]
4.5.1 with java 8 and 17 both tested using PGSQL or MSSQL as backends, still getting same error when logged into the admin GUI for any extended period of time.

**Comment by JoeFox82** [comment]
I did some tests and I think a found a "solution". The main problem are the tags in mirth. I opened the launcher with the java console and after some hours I got the awt exception with this info:

`ERROR 2024-06-18 05:55:06,223 [AWT-EventQueue-0] com.mirth.connect.client.ui.Frame: Error loading tags: java.lang.StackOverflowError
java.lang.StackOverflowError`

I deleted all tags in the channels and all tags in the settings menu. Then I did a restart and now - 35 hours after that - there is still no awt error.

Is it a bug or java problem?

**Reply by ramsoftmarc** [reply]
I'm having the same problem, but my Mirth doesn't have any tags.

**Reply by ramsoftmarc** [reply]
Error loading tags: java.lang.StackOverflowError
java.lang.StackOverflowError
at java.desktop/java.awt.AWTEventMulticaster.remove(Unknown Source)
at java.desktop/java.awt.AWTEventMulticaster.removeInternal(Unknown Source)

**Reply by JoeFox82** [reply]
That's a bit crazy. Since I deleted all the tags my mirth administrator launcher is running since 5 days without an awt error...

**Reply by ramsoftmarc** [reply]
Does anyone know what the EventMulticaster is?

**Reply by ramsoftmarc** [reply]
Your original screenshot shows the same error I got, but I'm not using channel tags.

**Reply by JoeFox82** [reply]
`[AWT-EventQueue-0] com.mirth.connect.client.ui.Frame: Error loading tags: java.lang.StackOverflowError java.lang.StackOverflowError`

Same tags error after 10 days, this time without using tags.

**Reply by ajadams2** [reply]
I'm having the same issue. No tags. 4.5.0 and using OpenJDK 12.0.2. I also have no channels running. It appears to occur if the leave the GUI open for a few days.

**Comment by tonygermano** [comment]
Lots of "me too" comments on here. This is a client-side error. It would be helpful if everyone with the error posted the following to see if any patterns emerge. It is highly unlikely that the server java version has anything to do with this.
- mirth version
- client-side java version
- client-side OS/architecture

**Reply by JoeFox82** [reply]
Tested with:
mirth version = 4.4.0 & 4.4.1 & 4.4.2 & 4.5.0
client-side java version = Java 17 (installed with mirth administrator launcher 1.4.1)
client-side OS/architecture = Windows Server 2016 / Windows Server 2019 (launcher is running on the same OS with mirth server)

**Reply by Dave7310** [reply]
I'm getting this error. been happening for about a week or so. 
Windows Server 2022 
Mirth 4.5.0
MCAL 1.4.1 bundled with Java 17

**Comment by ramsoftmarc** [comment]
Mirth 4.4.2
Running OpenJDK 64-Bit Server VM 17 on Windows 10, Windows 11 Enterprise, Windows Server 2012 R2 Standard, Windows Server 2016 Standard, Windows Server 2022 Standard, all with firebird.
It hasn't happened to Windows Server 2022 Datacenter Azure Edition yet.

**Comment by R1zwan** [comment]
Try settings your launcher heap size more then 2 gb. (4gb is more appropriate)

**Comment by prc9584** [comment]
Just having this occur again with 4Gb of heap space allocated, Mirth 4.5.2, Server 2022.
![image](https://github.com/user-attachments/assets/80e28ff3-b81b-4f49-a295-91bed25a8054)
![image](https://github.com/user-attachments/assets/207c70a7-dd38-4333-8c38-6432710604af)
![image](https://github.com/user-attachments/assets/6e0a125c-6652-479a-9125-90e803abaca3)
![image](https://github.com/user-attachments/assets/0db5da89-a81c-4591-9693-6834f3779952)
![image](https://github.com/user-attachments/assets/46b15603-e25e-446d-bc43-dae316ed1be6)

**Comment by stephrsh** [comment]
Hello,

same here... Is there any solution ?

Windows 2022 Std
Mirth 4.5.2
Microsoft OpenJDK 21.0.5

![Capture d’écran 2024-11-14 135948](https://github.com/user-attachments/assets/018820f6-dcb5-4002-adfc-c4ccac7bc3f0)
![Capture d’écran 2024-11-14 135756](https://github.com/user-attachments/assets/e0c82cd6-0906-4651-bb6e-5f82770e54e1)
![Capture d’écran 2024-11-14 135707](https://github.com/user-attachments/assets/3c25207d-6162-41a5-94e0-4a93ca22661a)

**Comment by SandhiyaKamaraj** [comment]
Hi all,

Facing same error. Did anyone found solution to fix this?

Mirth 4.4.1
Java - JDK11

Error loading tags: java.lang.StackOverflowError
java.lang.StackOverflowError
at java.desktop/java.awt.AWTEventMulticaster.remove(Unknown Source)
at java.desktop/java.awt.AWTEventMulticaster.removeInternal(Unknown Source)
at java.desktop/java.awt.AWTEventMulticaster.remove(Unknown Source)
at java.desktop/java.awt.AWTEventMulticaster.removeInternal(Unknown Source)

**Comment by justin-romano** [comment]
looks like an error causing an error for what its worth.
